torch/_tensor_str.py

```diff
@@ -115,20 +115,25 @@ def _number_format(tensor, min_sz=-1):
     if int_mode:
         if exp_max > prec + 1:
             format = '{{:11.{}e}}'.format(prec)
+            fmt_fn = lambda x: format.format(x)
             sz = max(min_sz, 7 + prec)
         else:
             sz = max(min_sz, exp_max + 1)
```

Comment:
good catch.  Maybe it makes sense to move the size calculation up, something like:
```
sz = max(min_sz, exp_max + 1 + include_decimal_int_mode)
format = '{:' + str(sz) + '.0f}'
fmt_fn = lambda x: format.format(x)
if include_decimal_int_mode:
    format_decimal = format + '.'
    fmt_fn = lambda x: format_decimal.format(x) if math.isfinite(x) else format.format(x)
 ```